src/Umbraco.Web/PropertyEditors/ValueConverters/MultiUrlPickerValueConverter.cs

```diff
@@ -15,77 +14,102 @@ namespace Umbraco.Web.PropertyEditors.ValueConverters
 
                     if (dto.Udi != null)
                     {
-                        type = dto.Udi.EntityType == Core.Constants.UdiEntityType.Media
-                            ? LinkType.Media
-                            : LinkType.Content;
-
-                        var content = type == LinkType.Media ?
-                             _publishedSnapshotAccessor.PublishedSnapshot.Media.GetById(preview, dto.Udi.Guid) :
-                             _publishedSnapshotAccessor.PublishedSnapshot.Content.GetById(preview, dto.Udi.Guid);
+                        switch (dto.Udi.EntityType)
+                        {
+                            case Core.Constants.UdiEntityType.Document:
+                                type = LinkType.Content;
+                                content = _publishedSnapshotAccessor.PublishedSnapshot.Content.GetById(preview, dto.Udi.Guid);
+                                break;
+                            case Core.Constants.UdiEntityType.Media:
+                                type = LinkType.Media;
+                                content = _publishedSnapshotAccessor.PublishedSnapshot.Media.GetById(preview, dto.Udi.Guid);
+                                break;
+                            default:
+                                // Skip unsupported entity types
+                                continue;
+                        }
 
-                        if (content == null || content.ContentType.ItemType == PublishedItemType.Element)
+                        if (content == null)
                         {
+                            // Skip unavailable content
                             continue;
                         }
-                        url = content.Url();
-                    }
 
-                    links.Add(
-                        new Link
+                        if (string.IsNullOrEmpty(name))
                         {
-                            Name = dto.Name,
-                            Target = dto.Target,
-                            Type = type,
-                            Udi = dto.Udi,
-                            Url = url + dto.QueryString,
+                            name = content.Name;
                         }
-                    );
+
+                        url = content.Url();
+                    }
+
+                    links.Add(new Link()
+                    {
+                        Name = name,
+                        Target = dto.Target,
+                        Type = type,
+                        Udi = dto.Udi,
+                        Content = content,
+                        Url = url + dto.QueryString
+                    });
+
+                    if (maxNumber > 0 && links.Count == maxNumber)
+                    {
+                        // Stop processing
+                        break;
+                    }
+                }
+
+                // Return the correct value type
+                if (maxNumber == 1)
```

Comment:
We're only returning here (and not in the foreach loop above) to make sure it always returns the correct value type. If all links are unpublished content, the list will be empty and it will correctly return `null`.